### ISSUE TITLE
can not find some header when complie with proto3 installed in my own path.

### ISSUE THREAD

**Issue by liwei-ch** [google/protobuf#176]
protobuf3 is installed in my path: /home/s/safe/protobuf3/
when i complie my project, find  some files lost: 
```
/home/s/safe/protobuf3/include/google/protobuf/map.h:35:40: error: google/protobuf/stubs/hash.h: No such file or directory
/home/s/safe/protobuf3/include/google/protobuf/map_field_inl.h:36:46: error: google/protobuf/stubs/shared_ptr.h: No such file or directory
/home/s/safe/protobuf3/src/google/protobuf/stubs/hash.h:40:20: error: config.h: No such file or directory
```

**Comment by xfxyjwf** [comment]
How did you install protobuf3?

**Comment by liwei-ch** [comment]
target=/home/s/safe/protobuf3
sudo -u cloud ./autogen.sh
sudo -u cloud ./configure --prefix=$target --oldincludedir=/home/s/safe/protobuf3/include
sudo -u cloud make
sudo -u cloud make check
sudo -u cloud make install

**Comment by liwei-ch** [comment]
to make it pass, i have to copy these files to target path.

**Comment by pherl** [comment]
This has been fixed in the v3.0.0-alpha-2 release

**Comment by AndreMiras** [comment]
FWIW, I also had the issue on Gentoo with `=dev-libs/protobuf-2.6.1-r3`, and installing `=dev-libs/protobuf-3.0.0_beta3_p1` solved it.
Thanks @pherl!
